### DISCUSSION TITLE
conditional in conky.config

### DISCUSSION THREAD

**Discussion by BETLOG** [brndnmtthws/conky#1550] [Q&A] [answered]
Is it possible to use conditionals inside conky.config?

If not... *feature request*!

I'm using X11forwarding to monitor 3 machines with copies of the same conky.conf copied to each machine, so i want to be able to use a hostname conditional to specify the correct  alignment = **inside conky.config ={}**

eg:
if $nodename == A
 alignment = 'top_left',
elif $nodename == B
 alignment = 'top_middle',
else
 alignment = 'top_right',

**Comment by ermshiperete** [comment]
There is a [`if_match`](https://conky.cc/variables#if_match) statement, so I believe the example could be written something like:

```
${if_match ${nodename} == 'A'}alignment = 'top_left'${else}${if_match ${nodename} == 'B'}alignment='top_middle'${else}alignment = ${top_right}${endif}${endif}
```

**Comment by Caellian** [comment, marked as answer]
You can't mix [$if_match](https://conky.cc/variables#if_match) in `conky.config` because it's pure Lua.

You can use instantly-invoked lambda pattern, or assign a merged table to `conky.config`. Latter is simpler and relatively common in conky configs. So long as by the end of config file evaluation `conky.config` is a table that contains the settings you want it should work fine.

In other words, you can incrementally build up the config in Lua:
```lua
conky.config = {} -- created by conky, even this line can be omitted
conky.config.alignment = "top_left"
```

And that would work just fine. I don't think you can use [`conky_parse(string)`](https://conky.cc/lua#conky_parse(string)) yet though because most variables aren't even populated yet when the config file is first parsed. BUT, you can use Lua equivalent that works for your system: `local hostname = io.popen("uname -n"):read("*a"):gsub("\n", "")`

So the final thing would be:

```lua
conky.config = {
  -- your standard configs here
  alignment = 'top_right', -- default case
}

local hostname = io.popen("uname -n"):read("*a"):gsub("\n", "")
if hostname == "A" then
  conky.config.alignment = "top_left"
elseif hostname == "B" then
  conky.config.alignment = "top_middle"
end

conky.text = [[ ... ]]
```

> [!CAUTION]
> `conky.config` is (currently?) only loaded once and mutations are ignored, changing its value after the initial load will not recreate conky window, update it or affect it in any way.

**Comment by EdyMarin0** [comment]
you could do a separate lua script that solves the if stqatements, then pull the results in conky.conf:

monitor.lua:
```
local monitor = conky_parse('${nodename}')
local alignment_position = ''
if monitor == A then
   alignment_position = 'top_left'
elseif monitor == B then
   alignment_position = 'top_middle'
else
  alignment_position = 'top_right'
end
```

conky.conf:
```
package.path = '~/.config/conky/scripts/?.lua' # path to the location of monitor.lua
local monitor= require 'monitor'

conky.config = {
alignment = monitor.alignment_position,
...
}
```

The monitor.lua gets evaluated once before conky.config is made, every time conky is run.
